### ISSUE TITLE
Change the spec of `objects.get_or_create` to only return the instance not a tuple

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida-core#4160]
Returning a boolean to indicate whether the instance was created is unnecessary as that can be deduced from the instance itself and it complicates the interface. If the instance is unstored it was per definition created as an unstored node could not have been retrieved from the database.

If we were to change the behavior and get rid of the special boolean swapping:
```group = Group.objects.get('label')```
for
```group = Group.objects.get_or_create('label')```
would become a lot easier.

**Comment by giovannipizzi** [comment]
As a note, this came from the `get_or_create` API of Django.

However, isn't `get_or_create` currently returning a stored node?
Also, can we do this before 2.0?

And finally, I have a small doubt on `group = Group.objects.get('label')` just returning an unstored group if you pass a random label string - this isn't what happens with the other `get` methods I think. Maybe if we find a different name, we solve all the issues.

**Comment by sphuber** [comment]
> As a note, this came from the `get_or_create` API of Django.

Interesting, wonder why they went down that road

> However, isn't `get_or_create` currently returning a stored node?
> Also, can we do this before 2.0?

No, this would be backwards incompatible, so if we think it is a good change, we should add a deprecation warning now. But maybe a normal one, so it does not show up everytime the code is called.

> And finally, I have a small doubt on `group = Group.objects.get('label')` just returning an unstored group if you pass a random label string

Why do you think this is the behavior? I looked in the code and see no evidence for it. This will currently raise a `NotExistent`. Ah, I see what you mean now. You were referring to my example. That would indeed just raise if it doesn't exist. That is the point, if you are you sure that it should exist, you use `get` just to load it. Otherwise, if you don't care if it exists and just want to get it, you use `get_or_create`. Alternatively, we could even break completely with Django and use `create=True` as a keyword argument to `get`.

**Comment by greschd** [comment]
> 
> 
> > As a note, this came from the `get_or_create` API of Django.
> 
> Interesting, wonder why they went down that road

It seems Django's `get_or_create` returns a _stored_ object, so it wouldn't be possible to know if it was newly created. See https://docs.djangoproject.com/en/3.1/ref/models/querysets/#get-or-create

**Comment by sphuber** [comment]
> TBH I'm not sure if this is a good change - knowing that `get_or_create` will return an unstored node and checking for that seems a little bit involved / not quite obvious.

But with the current API you *also* have to check? You just check an element of the tuple. So I don't see how that is a disadvantage of the proposed change

**Comment by greschd** [comment]
Yeah, but it's an obvious return value of the function, not an "obscure" attribute of the returned object.

The connection "unstored == newly created" is not totally obvious IMO, especially since a casual user might not have to care when exactly nodes are stored in AiiDA. It's completely reasonable to think `get_or_create` might always return stored nodes (as is done in Django, even though it doesn't make sense for us).

**Comment by sphuber** [comment]
Fair enough, but I think the approach of returning a tuple has downsides as well. It is not directly obvious that `get_or_create` returns a tuple instead of just the object. You would also have to read the docstring for this, or find out through trial and error. Moreover, I always find myself wondering which element of the tuple is the node and which is the boolean, which I think is another hint that shows that the method is not really intuitive

**Comment by greschd** [comment]
Yeah, I agree it's not great - especially since for most cases you don't care about whether it was new. But I don't think it's worth changing the interface over.

> Moreover, I always find myself wondering which element of the tuple is the node and which is the boolean

`namedtuple` to the rescue?
